assets/js/sidebar/sidebar-list.js

```diff
@@ -58,17 +58,22 @@ function renderSidebarNodeList (nodesByType, type) {
   nodeList.querySelectorAll('li a').forEach(anchor => {
     anchor.addEventListener('click', event => {
       const target = event.target
+      const listItem = target.closest('li')
+      const previousSection = nodeList.querySelector('.current-section')
 
-      // Allows only one sub-level to be open
-      if (anchor.matches('.summary') || (anchor.matches('.expand'))) {
-        const alreadyOpen = nodeList.querySelector('.docs.open')
-        if (alreadyOpen) {
-          alreadyOpen.classList.remove('open')
-        }
+      // Expand icon should not navigate
+      if (target.matches('.icon-expand')) {
+        event.preventDefault()
+        listItem.classList.toggle('open')
+        return
+      }
+
+      // Clear the previous current section
+      if (previousSection) {
+        previousSection.classList.remove('current-section')
       }
 
       if (anchor.matches('.expand')) {
-        const listItem = target.closest('li')
         listItem.classList.add('open')
```

Comment:
We shouldn't expand the item if we are navigating to another page (so that we don't expand and refresh immediately). A check like `anchor.pathname === window.location.pathname` should help :)

Reply:
Let's wait for @josevalim to confirm, but as far as I remember, we should now!